### PR TITLE
Tear-down

### PR DESCRIPTION
**Pull request by vokal-isaac**
- Take the repo as a positional argument instead of a flagged option.
- Added functions to remove IAM credentials, users, policy, and group.
- Exposed iam-credential-creation, travis-encryption, and iam-nuking at the library level.
- Expose the `--nuke` flag to completely nuke the IAM setup (with two confirmation prompts separated by a 5-second delay)

@projectweekend @WorkHorseIndustries @foresmac @adambain-vokal @Tuss4 @vhoof @scottferg @chillpop @brockboland Code review please?

### CODE REVIEW

**Comment by foresmac** [comment]
I don't see anything concerning about functionality. If I were nitpicking I'd point out that some of the indenting and line continuations don't quite follow our Python code standards, but I feel with all the changes here that would be better addressed in a separate PR.

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ If your AWS credentials are in the default profile, you can omit the `AWS_PROFIL
 
 With all the credentials in place, running
 ```
-$ s3same --repo some-repo-name --owner some-user --pro
+$ s3same some-repo-name --owner some-user --pro
```
Comment (brockboland) [inline README.md:42]: The usage line above (`Usage: s3same [OPTIONS] REPO`) implies that the repo name should come after the options, not before. Does it matter?

**Comment by brockboland** [comment]
Sounds good to me, but I'll have to rely on the systems folks for actual code review

File: bin/s3same [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
+    if not value or ctx.resilient_parsing:
+        return
+    if not click.confirm('Are you sure you want to nuke the whole s3same setup from IAM?'):
+        click.echo('Aborted.')
+        ctx.exit()
+    click.secho(
+            '\nDoing this will make any s3same-generated credentials invalid.\nThere is NO WAY to undo this.\n',
+            bold=True)
+    click.echo('Take a moment to think it over…   ', nl=False)
+    spinner = cycle(['|', '/', '-', '\\'])
```
Comment (WorkHorseIndustries) [inline bin/s3same:25]: I'm glad you added this.

File: s3same/iam.py [hunk 3]
```diff
@@ -29,16 +29,20 @@ def _policy_string(bucket):
...
         kwargs['Marker'] = response.get('Marker')
         if not response.get('IsTruncated') or kwargs['Marker'] is None:
-            return None
+            return
+
+def _find_policy(iam):
+    for policy in _all_pages(iam.list_policies, 'Policies', Scope='Local'):
+        if policy.get('PolicyName') == IAMName:
+            return policy
+    return None
```
Comment (WorkHorseIndustries) [inline s3same/iam.py:45]: don't need to explicitly return.  If you want to be super fancy

`next(
    (p for p in _all_pages(iam.list_policies, 'Policies', Scope='local' if p.get('PolicyName') == IAMName), 
    None)`

**Comment by WorkHorseIndustries** [comment]
Formatting issues aside, LGTM

File: s3same/iam.py (hunk 3 above)
Comment (vokal-isaac) [inline s3same/iam.py:45]: Heh, see [this](https://github.com/vokal/s3same/pull/1#discussion-diff-54206831).

File: bin/s3same (hunk 2 above)
Comment (vokal-isaac) [inline bin/s3same:25]: The spinner in particular or the double-confirm with the "No, really, think about it" delay?

File: README.md (hunk 1 above)
Comment (vokal-isaac) [inline README.md:42]: The order does not matter.  I don't have that level of control over the usage printout (and what I put in the README is the actual printout).

Comment (brockboland) [inline README.md:42]: Got it

Comment (brockboland) [inline README.md]: Got it

**Pull request merged**